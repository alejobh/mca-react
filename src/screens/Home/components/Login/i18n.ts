import i18next from 'i18next';

i18next.addResources('es', 'Login', {
  altWolox: 'Wolox',
  password: 'Password'
});
