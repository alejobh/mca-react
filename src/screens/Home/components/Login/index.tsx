/* eslint-disable react/react-in-jsx-scope */
import i18next from 'i18next';

import styles from './styles.module.scss';

function Login() {
  return (
    <div className={`row center middle full-width full-height ${styles.login}`}>
      {i18next.t('Login:logintitle')}
    </div>
  );
}

export default Login;
