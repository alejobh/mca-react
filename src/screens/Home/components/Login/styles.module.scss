@import '../../../../scss/variables/texts';

.login {
  font-size: $font-size-xlarge;
}
