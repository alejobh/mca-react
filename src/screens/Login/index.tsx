/* eslint-disable react/react-in-jsx-scope */
import { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useMutation } from 'react-query';
import { useHistory } from 'react-router-dom';
import i18next from 'i18next';

import WoloxLogo from 'assets/LogoWolox-Original.png';
import { login } from 'services/LoginService';
import Loading from 'components/Spinner/components/loading';

import { LoginProps, InputsValues, UserFieldIds } from '../../interfaces/interfaces';

import styles from './styles.module.scss';

function Login() {
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm<InputsValues>();
  const history = useHistory();
  const [notCredentials, setNotCredentials] = useState(false);

  const mutation = useMutation(login, {
    onError: error => {
      console.log(error);
      setNotCredentials(true);
    },
    onSuccess: data => {
      data.headers && window.localStorage.setItem('access-token', data.headers['access-token']);
      history.push('/');
    }
  });

  const onSubmit = (user: LoginProps) => mutation.mutate(user);

  return (
    <div className="column center middle full-width full-height">
      <form className={`m-bottom-2 ${styles.loginForm}`} onSubmit={handleSubmit(onSubmit)}>
        <img
          src={WoloxLogo}
          alt={`${i18next.t('Global:logoWolox')}`}
          className={`m-bottom-3 ${styles.logo}`}
        />
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label htmlFor={UserFieldIds.EMAIL} className={`m-bottom-2 ${styles.label}`}>
            {i18next.t('SignUp:email')}
          </label>
          <input
            type="email"
            {...register(UserFieldIds.EMAIL, {
              required: i18next.t('SignUp:required') as string,
              pattern: {
                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                message: i18next.t('SignUp:invalidFormat') as string
              }
            })}
            className={styles.input}
          />
          {errors.email && <span className={styles.inputError}>{errors.email.message}</span>}
        </div>
        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
          <label htmlFor={UserFieldIds.PASSWORD} className={`m-bottom-2 ${styles.label}`}>
            {i18next.t('Login:password')}
          </label>
          <input
            type="password"
            {...register(UserFieldIds.PASSWORD, {
              required: i18next.t('SignUp:required') as string,
              pattern: {
                value: /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[^\w\s\d])?.+$/,
                message: i18next.t('SignUp:validFormatPassword') as string
              },
              minLength: {
                value: 8,
                message: i18next.t('SignUp:minLenghtPassword') as string
              }
            })}
            className={styles.input}
          />
          {errors.password && <span className={styles.inputError}>{errors.password.message}</span>}
        </div>
        <button type="submit" className={`m-bottom-3 ${styles.loginButton}`}>
          {mutation.isLoading ? <Loading /> : 'Login'}
        </button>
        <div className={`m-bottom-3 ${styles.line}`} />
        <button type="button" className={styles.loginButtonSecondary} />
      </form>
      {notCredentials && <span className={styles.error}>Credenciales incorrectas</span>}
    </div>
  );
}

export default Login;
