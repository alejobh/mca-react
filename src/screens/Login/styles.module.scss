@import '../../scss/variables/colors';

.login-form {
  align-items: center;
  background-color: $white-smoke;
  border-top: 8px solid $cian;
  display: flex;
  flex-direction: column;
  max-width: 300px;
  padding: 28px 25px 24px;
  width: 100%;
}

.logo {
  height: 70px;
  width: 212px;
  object-fit: contain;
}

.input-container {
  width: 100%;
}

.input {
  height: 32px;
  width: 100%;
}

.input-error {
  color: $red;
  font-size: 10px;
}

.label {
  color: $black;
  font-size: 16px;
  line-height: 20px;
}

.login-button {
  background-color: $yellow-green;
  border-radius: 5px;
  color: $white;
  font-size: 16px;
  height: 32px;
  width: 100%;

  &-secondary {
    background-image: url('../../assets/sign-up.png');
    height: 32px;
    width: 100%;
  }
}

.line {
  background-color: $light-grey;
  height: 2px;
  width: 100%;
}

.no-error {
  display: none;
}

.error {
  color: $red;
  display: block;
  font-size: 20px;
}
