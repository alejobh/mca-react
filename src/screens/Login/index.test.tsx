import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from 'react-query';

import Login from './index';

const queryClient = new QueryClient();

describe('Login component', () => {
  test('matches last snapshot', () => {
    const component = render(
      <QueryClientProvider client={queryClient}>
        <Login />
      </QueryClientProvider>
    );

    expect(component).toMatchSnapshot();
    userEvent.type(screen.getByLabelText('SignUp:email'), 'test@gmail.com');
    userEvent.type(screen.getByLabelText('Login:password'), 'wolox1190');
  });
});
