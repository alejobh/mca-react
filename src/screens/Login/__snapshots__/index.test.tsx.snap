// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches last snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="column center middle full-width full-height"
      >
        <form
          class="m-bottom-2 loginForm"
        >
          <img
            alt="Global:logoWolox"
            class="m-bottom-3 logo"
            src="LogoWolox-Original.png"
          />
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="email"
              id="email-label"
            >
              SignUp:email
            </label>
            <input
              aria-labelledby="email-label"
              class="input"
              name="email"
              type="email"
            />
          </div>
          <div
            class="column start m-bottom-3 inputContainer"
          >
            <label
              class="m-bottom-2 label"
              for="password"
              id="password-label"
            >
              Login:password
            </label>
            <input
              aria-labelledby="password-label"
              class="input"
              name="password"
              type="password"
            />
          </div>
          <button
            class="m-bottom-3 loginButton"
            type="submit"
          >
            Login
          </button>
          <div
            class="m-bottom-3 line"
          />
          <button
            class="loginButtonSecondary"
            type="button"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="column center middle full-width full-height"
    >
      <form
        class="m-bottom-2 loginForm"
      >
        <img
          alt="Global:logoWolox"
          class="m-bottom-3 logo"
          src="LogoWolox-Original.png"
        />
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="email"
            id="email-label"
          >
            SignUp:email
          </label>
          <input
            aria-labelledby="email-label"
            class="input"
            name="email"
            type="email"
          />
        </div>
        <div
          class="column start m-bottom-3 inputContainer"
        >
          <label
            class="m-bottom-2 label"
            for="password"
            id="password-label"
          >
            Login:password
          </label>
          <input
            aria-labelledby="password-label"
            class="input"
            name="password"
            type="password"
          />
        </div>
        <button
          class="m-bottom-3 loginButton"
          type="submit"
        >
          Login
        </button>
        <div
          class="m-bottom-3 line"
        />
        <button
          class="loginButtonSecondary"
          type="button"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
